### PR TITLE
add Span to Envelope converter

### PR DESCRIPTION
**Pull request by markwolff**

### CODE REVIEW

File: src/utils/spanUtils.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+function createInProcData(span: ReadableSpan): RemoteDependencyData {
+  const data = createDependencyData(span);
+  data.type = 'InProc';
+  data.success = true;
+  return data;
+}
+
+export function readableSpanToEnvelope(
```
Comment (lzchen) [inline src/utils/spanUtils.ts:144]: What is actually going to be calling this? AzureMonitorSpanExporter that's going to extend from `AzureMonitorBaseExporter` correct?

File: src/utils/spanUtils.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      break;
+    default:
+      // never
+      if (logger) {
+        logger.error(`Unsupported span kind ${span.kind}`);
+      }
+      throw new Error(`Unsupported span kind ${span.kind}`);
+  }
+
+  envelope.data.baseData = { ...data, properties };
```
Comment (lzchen) [inline src/utils/spanUtils.ts:180]: Where is AI specific context information stored? (ai.cloud.role, ai.internal.sdkversion, etc)

File: src/utils/spanUtils.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  properties['_MS.links'] = links;
+
+  return properties;
+}
+
+function createDependencyData(span: ReadableSpan): RemoteDependencyData {
+  const data = new RemoteDependencyData();
+  data.name = span.name;
+  data.id = `|${span.spanContext.traceId}.${span.spanContext.spanId}.`;
```
Comment (xiao-lix) [inline src/utils/spanUtils.ts:62]: Is the last "." added on purpose? curious what does it concat with?

File: test/utils/spanUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        assert.strictEqual(envelope.name, 'Microsoft.ApplicationInsights.RemoteDependency');
+        assert.deepStrictEqual(envelope.data?.baseType, 'RemoteDependencyData');
+
+        assert.strictEqual(envelope.iKey, 'ikey');
+        assert.ok(envelope.time);
+        assert.ok(envelope.ver);
+        assert.ok(envelope.data);
+
+        assert.deepStrictEqual(envelope.tags, expectedTags);
+        assert.deepStrictEqual(envelope.data?.baseData?.properties, expectedProperties);
```
Comment (xiao-lix) [inline test/utils/spanUtils.test.ts]: Interesting, didn't know "?" can be used this way, does it stop when envelope.data is undefined?

File: src/utils/spanUtils.ts (hunk 3 above)
Comment (markwolff) [reply src/utils/spanUtils.ts:62]: It is required for parentId to get picked up properly

File: test/utils/spanUtils.test.ts (hunk 4 above)
Comment (markwolff) [reply test/utils/spanUtils.test.ts]: Yep! It was added to TS3 a few months ago

File: src/utils/spanUtils.ts (hunk 2 above)
Comment (markwolff) [reply src/utils/spanUtils.ts:180]: Will add in separate PR

File: src/utils/spanUtils.ts (hunk 1 above)
Comment (markwolff) [reply src/utils/spanUtils.ts:144]: yes

**Review by lzchen — APPROVED** [review]

**Pull request merged**